examples/complete/main.tf

```diff
@@ -115,19 +115,18 @@ resource "ibm_is_security_group" "sg1" {
   vpc  = ibm_is_vpc.example_vpc.id
 }
 
-resource "ibm_is_virtual_endpoint_gateway" "pgvpe" {
-  name = "${var.prefix}-vpe-to-pg"
-  target {
-    crn           = module.postgresql_db.crn
-    resource_type = "provider_cloud_service"
-  }
-  vpc = ibm_is_vpc.example_vpc.id
-  ips {
-    subnet = ibm_is_subnet.testacc_subnet.id
-    name   = "${var.prefix}-pg-access-reserved-ip"
-  }
-  resource_group  = module.resource_group.resource_group_id
-  security_groups = [ibm_is_security_group.sg1.id]
+module "vpe" {
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-vpe-module?ref=v2.4.0"
+  prefix = "${var.prefix}-vpe-to-pg"
+  cloud_service_by_crn = [
+    {
+      name = "${var.prefix}-postgres"
+      crn  = module.postgresql_db.crn
+    },
+  ]
+  vpc_id             = ibm_is_vpc.example_vpc.id
+  resource_group_id  = module.resource_group.resource_group_id
+  security_group_ids = [ibm_is_security_group.sg1.id]
```

Comment:
Isn't subnet_zone_list a required parameter for this module? Technically it is optional, but if it defaults, then as far as I can see, it is not connected to anything.

Reply:
Now it passes a default (empty list) value.

testacc_subnet is created, oddly only in one zone (I am not sure if that is enough for connectivity from all 3 zones, but it is certainly not resilient). At no point is a reserved IP allocated from this subnet and pointed to the gateway. So as far as I can see, it is still not connected to anything.

I am raising an issue against the VPE gateway documentation, because it is really not clear what is required.

Reply:
A little further investigation, there is already an `ibm_is_subnet`, it seems an `ibm_is_subnet_reserved_ip` is required with `subnet` binding to `target` (VPE gateway) properties. The is managed by the vpe module at https://github.com/terraform-ibm-modules/terraform-ibm-vpe-module/blob/main/main.tf#L69

This replaces the old `ips {}` block on the VPE gateway definition in more appropriate form (one that can be modified later).

Maybe we need a less complex form of https://github.com/terraform-ibm-modules/terraform-ibm-vpe-module/blob/main/examples/default/main.tf#L91, since this example always create the VPC, no conditional logic is required.

Reply:
Hi @shemau , [here](https://github.com/terraform-ibm-modules/terraform-ibm-icd-postgresql/pull/214#:~:text=So%20maybe%20delete%20out%20subnet%20and%20just%20use%20the%20subnet%20in%20the%20VPC%3F), are you asking to delete the testacc_subnet which is being created in the complete example of postgres and instead add the subnet in the VPC created as example_vpc in the complete example of postgres? The testacc_subnet is being created in the example_vpc.